### PR TITLE
[16.0][IMP] stock_release_channel: Add a company setting to enable/disable recomputation of channel on release

### PR DESCRIPTION
**Pull request by QuocDuong1306**

### CODE REVIEW

**Review by santostelmo — APPROVED** [review]

File: stock_release_channel/views/res_company.xml [hunk 1]
```diff
@@ -0,0 +1,21 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<odoo>
+    <record model="ir.ui.view" id="view_company_form">
+        <field name="model">res.company</field>
+        <field name="inherit_id" ref="base.view_company_form" />
```
Comment (jbaudoux) [inline stock_release_channel/views/res_company.xml]: Don't put this directly on the company view but define a res config setting related to res company. Look at other modules, there are many examples

File: stock_release_channel/static/description/index.html [hunk 2]
```diff
@@ -1,3 +1,4 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (jbaudoux) [inline stock_release_channel/static/description/index.html]: please drop this change

**Review by jbaudoux — CHANGES_REQUESTED** [review]

**Comment by jbaudoux** [comment]
ping @QuocDuong1306

**Comment by jbaudoux** [comment]
@QuocDuong1306 Can you fix the review remarks?

**Comment by QuocDuong1306** [comment]
Hello @jbaudoux , I updated the PR

File: stock_release_channel/models/stock_move.py [hunk 3]
```diff
@@ -16,7 +16,9 @@ def release_available_to_promise(self):
         # As moves can be merged (and then unlinked), we should ensure
         # they still exist.
         moves = self.exists()
-        moves.picking_id.assign_release_channel()
+        company = self.env.company
```
Comment (jbaudoux) [inline stock_release_channel/models/stock_move.py]: There is a company field on the move, you need to use that one
```suggestion
        for company, imoves in groupby(moves, key=itemgetter("company_id"))
            if company.recompute_channel_on_pickings_at_release:
                moves = self.env['stock.move'].concat(*imoves)
                moves.picking_id.assign_release_channel()
```

**Review by jbaudoux — CHANGES_REQUESTED** [review]

File: stock_release_channel/models/stock_move.py (hunk 3 above)
Comment (QuocDuong1306) [reply stock_release_channel/models/stock_move.py]: Thank @jbaudoux , I updated the PR

**Review by jbaudoux — APPROVED** [review]

**Comment by jbaudoux** [comment]
We need a test for `recompute_channel_on_pickings_at_release`

**Comment by QuocDuong1306** [comment]
Hello @jbaudoux , added test

**Review by jbaudoux — APPROVED** [review]

**Pull request merged**